tad/core/db.py

```diff
@@ -43,21 +56,59 @@ def init_db():
 
     with Session(get_engine()) as session:
         if get_settings().ENVIRONMENT == "demo":
+            if get_settings().TRUNCATE_TABLES:
+                truncate_tables(session)
             logger.info("Creating demo data")
-
-            user = session.exec(select(User).where(User.name == "Robbert")).first()
-            if not user:
-                user = User(name="Robbert", avatar=None)
-                session.add(user)
-
-            status = session.exec(select(Status).where(Status.name == "Todo")).first()
-            if not status:
-                status = Status(name="Todo", sort_order=1)
-                session.add(status)
-
-            task = session.exec(select(Task).where(Task.title == "First task")).first()
-            if not task:
-                task = Task(title="First task", description="This is the first task", sort_order=1, status_id=status.id)
-                session.add(task)
-            session.commit()
+            remove_old_demo_objects(session)
+            add_demo_users(session, ["default user"])
+            demo_statuses = add_demo_statuses(session, ["todo", "review", "in_progress", "done"])
+            add_demo_tasks(session, demo_statuses[0], 3)
     logger.info("Finished initializing database")
+
+
+def truncate_tables(session: Session) -> None:
```

Comment:
This is rather scary code to have, can we guard it with something like:

```python
if get_settings().ENVIRONMENT == "prod":
    return
```

Reply:
I think no one needed this besides me and there are others ways for what I need. I will delete this function as it undoubtedly will create an unintended database wipe in the future.

Reply:
It would be better to guard it in the tad/core/config as a model_validator:

```
    @model_validator(mode="after")
    def _enforce_database_cleanup_rules(self: SelfSettings) -> SelfSettings:
        if self.ENVIRONMENT == "production" and self.TRUNCATE_TABLES:
            raise SettingsError("TRUNCATE_TABLES=True is not supported in production")
        return self
```

Reply:
Dat is wel heel cool. Die code zal ik zeker onthouden. Voor nu heb ik de truncate er uit gelaten. Het is niet perse nodig en daarmee voorkomen we dat het uberhaubt fout kan gaan ;)